src/App.js

```diff
@@ -1200,11 +1099,10 @@ render() {
                   />
                 </div>
               )
-            }else if(this.state.isVendor&&this.state.isVendor.isAllowed){
```

Comment:
Why can we remove `this.state.isVendor.isAllowed`?

Reply:
Good catch, we can't :)

I was originally working on abstracting the VendingMachine contract, but it got a little complex and out of scope. I've reverted this line